src/InteractionHandler.php

```diff
@@ -4,76 +4,144 @@
 
 namespace Exan\Fenrir;
 
+use Closure;
 use Exan\Fenrir\Component\Button\InteractionButton;
 use Exan\Fenrir\Constants\Events;
 use Exan\Fenrir\Enums\Parts\InteractionTypes;
 use Exan\Fenrir\Interaction\ButtonInteraction;
 use Exan\Fenrir\Interaction\CommandInteraction;
 use Exan\Fenrir\Parts\ApplicationCommand;
 use Exan\Fenrir\Rest\Helpers\Command\CommandBuilder;
+use Exan\Fenrir\Rest\Helpers\Command\QueuedCommand;
 use Exan\Fenrir\Websocket\Events\InteractionCreate;
 use Exan\Fenrir\Websocket\Events\Ready;
+use Throwable;
 
 class InteractionHandler
 {
     private FilteredEventEmitter $commandListener;
     private FilteredEventEmitter $buttonListener;
+    private string $applicationId;
+    private bool $devMode = false;
 
     /** @var array<string, callable> */
     private array $handlersCommand = [];
 
     /** @var array<string, callable> */
     private array $handlersButton = [];
 
-    private bool $devMode = false;
+    /** @var QueuedCommand[] */
+    private array $commandQueue = [];
+
+    /** @var string[] */
+    private array $guildsToCheck = [];
 
     public function __construct(private Discord $discord, private ?string $devGuildId = null)
     {
         if (!is_null($this->devGuildId)) {
             $this->devMode = true;
         }
+
+        $this->discord->gateway->events->once(
+            Events::READY,
+            function (Ready $ready) {
+                $this->applicationId = $ready->user->id;
+
+                $compareCommands = function (array $commands, bool $globalCommands) {
+                    foreach ($this->commandQueue as $queuedCommand) {
+                        if ($globalCommands && ($queuedCommand->guildId !== null)) {
+                            return;
+                        }
+
+                        $needsRegistered = true;
```

Comment:
Apart from me not understanding why you need this in the first place, `$needsRegistration`.